orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/providers/azure/AzureServerGroupCreator.groovy

```diff
@@ -38,6 +39,14 @@ class AzureServerGroupCreator implements ServerGroupCreator {
       operation.credentials = operation.account
     }
 
+    def bakeStage = getPreviousStageWithImage(stage, operation.region)
+
+    if (bakeStage) {
+      operation.image.isCustom = true
+      operation.image.uri = bakeStage?.context?.imageId
```

Comment:
null dereferencing with ? shouldn't be necessary as you've already asserted bakeStage is not null